snuba/query/snql/parser.py

```diff
@@ -129,8 +129,7 @@
     function_call         = function_name open_paren parameters_list? close_paren (open_paren parameters_list? close_paren)? (space* "AS" space* string_literal)?
     simple_term           = quoted_literal / numeric_literal / column_name
     literal               = ~r"[a-zA-Z0-9_\.:-]+"
-    quoted_literal        = "'" quoted_text "'"
-    quoted_text           = ~r"[^']*"
+    quoted_literal        = ~r"((?<!\\)')(.(?!(?<!\\)'))*.?'"
```

Comment:
Why do you need to match `.?` right before the trailing quote ? Isn't that covered by the previous `(.(?!(?<!\\)'))*` ?

Reply:
No because the backref doesn't match that character, so we need to handle that character that appears before the ' explicitly.